src/scenic/core/regions.py

```diff
@@ -1546,77 +1585,85 @@ class BoxRegion(MeshVolumeRegion):
-        position: An optional position, which determines where the center of the region will be.
-        rotation: An optional Orientation object which determines the rotation of the object in space.
-        orientation: An optional vector field describing the preferred orientation at every point in
-          the region.
-        tolerance: Tolerance for collision computations.
-        engine: Which engine to use for mesh operations. Either "blender" or "scad".
+    Parameters are the same as `MeshVolumeRegion`, with the exception of the ``mesh``
+    parameter which is excluded.
     """
-    def __init__(self, name=None, dimensions=None, position=None, rotation=None, orientation=None,
-                 tolerance=1e-8, onDirection=None, engine="blender"):
+    def __init__(self, **kwargs):
         sphere_mesh = trimesh.creation.icosphere(radius=1)
-        super().__init__(mesh=sphere_mesh, name=name, position=position, rotation=rotation, dimensions=dimensions, \
-            orientation=orientation, tolerance=tolerance, onDirection=onDirection, engine=engine)
+        super().__init__(mesh=sphere_mesh, **kwargs)
 
     @cached_property
     def isConvex(self):
         return True
 
     def sampleGiven(self, value):
-        return SpheroidRegion(name=self.name, dimensions=value[self.dimensions],
-            position=value[self.position], rotation=value[self.rotation], \
-            orientation=self.orientation, tolerance=self.tolerance, engine=self.engine)
+        return SpheroidRegion(dimensions=value[self.dimensions],
+            position=value[self.position], rotation=value[self.rotation],
+            orientation=value[self.orientation], tolerance=self.tolerance,
+            engine=self.engine, name=self.name)
+
+    def evaluateInner(self, context):
+        dimensions = valueInContext(self.dimensions, context)
+        position = valueInContext(self.position, context)
+        rotation = valueInContext(self.rotation, context)
+        orientation = valueInContext(self.orientation, context)
+
+        return SpheroidRegion(dimensions=dimensions, position=position,
+            rotation=rotation, orientation=orientation, tolerance=self.tolerance,
+            engine=self.engine, name=self.name)
 
 class PolygonalFootprintRegion(Region):
     """ Region that contains all points in a polygonal footprint, regardless of their z value. 
 
     This region cannot be sampled from, as it has infinite height and therefore infinite volume.
     
     Args:
-        polygon: A ``MultiPolygon`` that defines the footprint of this region.
+        polygons: A ``MultiPolygon`` that defines the footprint of this region.
         name: An optional name to help with debugging.
     """
-    def __init__(self, polygon, name=None):
-        if not isinstance(polygon, MultiPolygon):
+    def __init__(self, polygons, name=None):
+        if not isinstance(polygons, MultiPolygon):
```

Comment:
This check should be removed if you're trying to allow single polygons as below. (In which case you should just keep the parameter name as `polygon`, consistent with `PolygonalRegion`. The docstring will also need to be updated.)